### PR TITLE
ext: Import Atmel SAM3U header files from ASF library

### PR DESCRIPTION
**Pull request by xingrz**
Atmel Software Framework (ASF) provides a set of low-level header files that give access to different hardware peripherals of Atmel's ICs.

Origin: Atmel SAM3U Series Device Support (1.0.49)
License: SAM Software Package License
URL: http://packs.download.atmel.com/Atmel.SAM3U_DFP.1.0.49.atpack
Purpose: Introduction of ASF for the SAM3U series.
Maintained-by: External

### CODE REVIEW

**Comment by nandojve** [comment]
Hi @xingrz ,

Make sure you are following the documentation which will be soon introduced by #31.

**Comment by nandojve** [comment]
Hi @stephanosio ,

There is no issues to add code `license: bsd-atmel` right?

File: asf/sam/include/sam3u/README [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# Copyright (c) 2023 Chen Xingyu <hi@xingrz.me>
+# SPDX-License-Identifier: Apache-2.0
```
Comment (stephanosio) [inline asf/sam/include/sam3u/README:2]: README file does not need a copyright notice.

**Review by stephanosio — CHANGES_REQUESTED** [review]
The commit message says:

```
License: Apache-2.0
```

This is incorrect. The external source code being added is clearly not Apache-2.0 licensed.

**Comment by stephanosio** [comment]
> There is no issues to add code `license: bsd-atmel` right?

Although the "SAM Software Package License" is essentially a variant/mix of the 2-clause and 3-clause BSD licenses, it is not one of the [OSI approved licenses](https://opensource.org/license/).

@nashif Have we accepted a variant of the BSD license that does not correspond to one of the OSI approved BSD license variants?

File: asf/sam/include/sam3u/README (hunk 1 above)
Comment (xingrz) [reply asf/sam/include/sam3u/README:2]: Okay. I have no idea. I just follow what c1c4410389eb51ab43a0da271812f3c5328fde17 is did.

**Review by nandojve — APPROVED** [review]
LGTM!

I'll not merge because I'm still waiting a position about licenses. Besides that, those are BSD like and I think there is no restriction about them.

**Comment by nandojve** [comment]
Can we merge this since seems not to be a problem?

**Comment by henrikbrixandersen** [comment]
> > There is no issues to add code `license: bsd-atmel` right?
> 
> Although the "SAM Software Package License" is essentially a variant/mix of the 2-clause and 3-clause BSD licenses, it is not one of the [OSI approved licenses](https://opensource.org/license/).
> 
> @nashif Have we accepted a variant of the BSD license that does not correspond to one of the OSI approved BSD license variants?

Not to my knowledge. I know that the board has previously rejected a slightly modified Apache-2.0 license text since it was not OSI approved (see https://github.com/zephyrproject-rtos/zephyr/pull/19510).

**Comment by nandojve** [comment]
I found this link [bsd-atmel](https://scancode-licensedb.aboutcode.org/bsd-atmel.html). It seems that it is a permissive license and there is no reason to not use. Is it possible to send this topic to TSC have a definitive vote about this license or better way is directly send relative info direct to [licensereview@lists.zephyrproject.org](mailto:licensereview@lists.zephyrproject.org) ?

**Comment by stephanosio** [comment]
> I found this link [bsd-atmel](https://scancode-licensedb.aboutcode.org/bsd-atmel.html). It seems that it is a permissive license and there is no reason to not use.

While I personally agree with you, the documented process requires an OSI-approved license, which `bsd-atmel` currently is not.

> Is it possible to send this topic to TSC have a definitive vote about this license or better way is directly send relative info direct to [licensereview@lists.zephyrproject.org](mailto:licensereview@lists.zephyrproject.org) ?

You can create an "External Source Code" issue for this in the Zephyr main repository and add "TSC" label so that this can be discussed in the next TSC meeting. I cannot say the outcome will be favourable given the current documented process; but, doing so will at least raise the awareness of this issue.

**Comment by nandojve** [comment]
Opened https://github.com/zephyrproject-rtos/zephyr/issues/57400 to request exception for `bsd-atmel` license,

**Comment by MyGh64605** [comment]
This Michael Sherwood of Microchip. We are reviewing our options internally at this time.

**Comment by nandojve** [comment]
> Sorry, but as far as the [documented process](https://docs.zephyrproject.org/latest/contribute/external.html#submission-and-review-process) is concerned, "SAM Software Package License" (aka. `bsd-atmel` license) is not an [Open Source Initiative (OSI)](https://opensource.org/licenses/) approved license and cannot be contributed to Zephyr.
> 
> In order for this to be eligible, either the OSI will need to approve and list the `bsd-atmel` license, or Microchip will need to re-license these files under one of the OSI approved licenses.

@NhMchp ,

Is it possible to add bsd-atmel to be listed in the OSI? This will allow the community to move and add other devices.

Or Is it possible to release a new DFP that only changes the license to one of those from OSI like bsd only or apache 2.0?

**Pull request closed without merging**